sycl-fusion/common/include/Kernel.h

```diff
@@ -193,6 +177,39 @@ class Indices {
   size_t Values[Size];
 };
 
+///
+/// Model SYCL/OpenCL kernel attributes that are handled by the JIT.
+enum SYCLKernelAttributeKind { Invalid, ReqdWorkGroupSize, WorkGroupSizeHint };
+
+///
+/// Describe a SYCL/OpenCL kernel attribute by its kind and values.
+struct SYCLKernelAttribute {
+  static constexpr auto ReqdWorkGroupSizeName = "reqd_work_group_size";
+  static constexpr auto WorkGroupSizeHintName = "work_group_size_hint";
+
+  SYCLKernelAttributeKind Kind;
+  Indices Values;
+
+  SYCLKernelAttribute() : Kind(Invalid) {}
+  SYCLKernelAttribute(SYCLKernelAttributeKind Kind, const Indices &Values)
+      : Kind(Kind), Values(Values) {}
+
+  const char *getName() const {
+    switch (Kind) {
+    case ReqdWorkGroupSize:
+      return ReqdWorkGroupSizeName;
+    case WorkGroupSizeHint:
+      return WorkGroupSizeHintName;
+    default:
+      return nullptr;
```

Comment:
I don't like the idea of returning `nullptr` on error. Can we `assert(Kind != Invalid)` on entry?